src/test/resources/org/jenkinsci/plugins/ewm/casc/configuration-as-code.yaml

```diff
@@ -0,0 +1,74 @@
+jenkins:
+  systemMessage: "Jenkins configured automatically by Jenkins Configuration as Code Plugin"
+  numExecutors: 5
+  scmCheckoutRetryCount: 2
+  mode: NORMAL
+  nodeProperties:
+    - exwsNodeConfigurationDiskPools:
+        nodeDiskPools:
+          - diskPoolRefId: "master-node-id"
+            nodeDisks:
+              - diskRefId: "master-node-disk"
+                nodeMountPoint: "/tmp/master-node"
+
+## TODO [Martin]: The following code crashes JCasC for reasons I do not understand
```

Comment:
What's the output?

Reply:
If the change to ConfiguredWithCode is done properly in https://github.com/imaffe/external-workspace-manager-plugin/pull/5 this passes unit tests @martinda

Reply:
@casz The error message given is:
```[INFO] Running org.jenkinsci.plugins.ewm.casc.ConfigAsCodeTest
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.507 s <<< FAILURE! - in org.jenkinsci.plugins.ewm.casc.ConfigAsCodeTest
[ERROR] org.jenkinsci.plugins.ewm.casc.ConfigAsCodeTest  Time elapsed: 2.502 s  <<< FAILURE!
java.lang.AssertionError: Unexpected exception 
	at io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule.before(JenkinsConfiguredWithCodeRule.java:38)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:551)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: Single entry map expected to configure a hudson.slaves.ComputerLauncher
	at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.configureMapping(HeteroDescribableConfigurator.java:270)
	at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.preConfigure(HeteroDescribableConfigurator.java:252)
	at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.configure(HeteroDescribableConfigurator.java:83)
	at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.configure(HeteroDescribableConfigurator.java:55)
	at io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator.tryConstructor(DataBoundConfigurator.java:131)
	at io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator.instance(DataBoundConfigurator.java:74)
	at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:268)
	at io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator.configure(DataBoundConfigurator.java:80)
	at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$doConfigure$16668e2$1(HeteroDescribableConfigurator.java:278)
	at io.vavr.CheckedFunction0.lambda$unchecked$52349c75$1(CheckedFunction0.java:201)
	at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.doConfigure(HeteroDescribableConfigurator.java:278)
	at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$null$2(HeteroDescribableConfigurator.java:86)
	at io.vavr.control.Option.map(Option.java:373)
	at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.lambda$configure$3(HeteroDescribableConfigurator.java:86)
	at io.vavr.Tuple2.apply(Tuple2.java:239)
	at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.configure(HeteroDescribableConfigurator.java:83)
	at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.check(HeteroDescribableConfigurator.java:92)
	at io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator.check(HeteroDescribableConfigurator.java:55)
	at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:345)
	at io.jenkins.plugins.casc.BaseConfigurator.check(BaseConfigurator.java:288)
	at io.jenkins.plugins.casc.ConfigurationAsCode.lambda$checkWith$6(ConfigurationAsCode.java:672)
	at io.jenkins.plugins.casc.ConfigurationAsCode.invokeWith(ConfigurationAsCode.java:634)
	at io.jenkins.plugins.casc.ConfigurationAsCode.checkWith(ConfigurationAsCode.java:672)
	at io.jenkins.plugins.casc.ConfigurationAsCode.configureWith(ConfigurationAsCode.java:656)
	at io.jenkins.plugins.casc.ConfigurationAsCode.configureWith(ConfigurationAsCode.java:555)
	at io.jenkins.plugins.casc.ConfigurationAsCode.configure(ConfigurationAsCode.java:525)
	at io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule.before(JenkinsConfiguredWithCodeRule.java:35)
	... 5 more```

Reply:
Indentation errors and the `jnlp: `line cause this. https://github.com/imaffe/external-workspace-manager-plugin/pull/5#issuecomment-511001150

Reply:
But do we need these lines? Looks like a similar config is defined beginning at line 34.

Reply:
I don't think `jnlp` is needed, I copied some of this code from either 1) examples I found, or 2) from a JCasC dump itself (but I ignored the warnings about the risk of using JCasC dumps verbatim).

Reply:
This review comment is likely resolved, then. The current commit has the `jnlp:` line omitted.

Reply:
Can't we remove the commented lines from [14-32]? :)

Reply:
It's at @martinda discretion to decide to use either the commented lines and delete [34-47], or to delete the commented lines and keep the rest the same.